src/charm.py

```diff
@@ -65,10 +72,31 @@ def _get_kubeconfig_status(self):
                     return True
         return False
 
+    def _install_cli_resources(self):
+        try:
+            cli_clients_path = Path("/usr/local/bin")
+            for rsc in RESOURCES:
+                path = self.model.resources.fetch(rsc)
+                shutil.copy(path, cli_clients_path)
+                os.chmod(cli_clients_path / rsc, 0o755)
```

Comment:
i'm not saying it's wrong -- but i'm used to seeing the resources as tgz instead of the extracted binaries.  Is there a reason why you've chosen this path? 

also -- should i assume that the charm only supports amd64 and for that reason we're only providing that arch's binary in the resource?

Reply:
I don't think there's a good reason to bundle the extracted binaries, I've mainly been driving the logic from the bash script I've created. Nevertheless, I think you have a great idea here because sending the `tgz` file will save storage/network usage.

As for the charm architectures, Cilium supports a handful of them. I think including the most commonly used is the way to go (e.g. `arm64` and `amd64`). What do you think?

Reply:
so, i can point you at how i handled this with [containerd](https://github.com/charmed-kubernetes/charm-containerd/pull/84)

The resource supports a "multi-arch" tgz for each binary.